### ISSUE TITLE
spack view command produces merge blocked by file error for binutils

### ISSUE THREAD

**Issue by mcfadden8** [spack/spack#9029]
I ran the following command with the ```commit 63004e3de1473254c3b2a75f1e06afaabdacd7dd (HEAD -> develop, origin/develop, origin/HEAD)``` version of spack:

```spack view symlink ~/.bin/darwin-highsierra-x86_64/ binutils```

in order to populate the bin, include, lib, man, and share directories with symbolic links to items associated with the ```binutils``` installation.  This command did not complete successfully.

### Steps to reproduce the issue (and error message)
I did the following on a clean system:

```console
$ git clone git@github.com:spack/spack.git
$ spack install binutils
$ spack -sd view symlink ~/.bin/darwin-highsierra-x86_64/ binutils
lib/spack/spack/config.py:630 ==> Reading config file /Users/mcfadden8/spack/etc/spack/defaults/modules.yaml
lib/spack/spack/config.py:630 ==> Reading config file /Users/mcfadden8/spack/etc/spack/defaults/config.yaml
lib/spack/llnl/util/lock.py:303 ==> READ LOCK: /Users/mcfadden8/spack/opt/spack/.spack-db/lock[0:0] [Acquiring]
lib/spack/llnl/util/lock.py:303 ==> READ LOCK: /Users/mcfadden8/spack/opt/spack/.spack-db/lock[0:0] [Acquired]
lib/spack/spack/config.py:630 ==> Reading config file /Users/mcfadden8/spack/etc/spack/defaults/repos.yaml
lib/spack/llnl/util/lock.py:303 ==> READ LOCK: /Users/mcfadden8/spack/opt/spack/.spack-db/lock[0:0] [Released]
Traceback (most recent call last):
  File "/Users/mcfadden8/spack/bin/spack", line 54, in <module>
    sys.exit(spack.main.main())
  File "/Users/mcfadden8/spack/lib/spack/spack/main.py", line 653, in main
    return _invoke_command(command, parser, args, unknown)
  File "/Users/mcfadden8/spack/lib/spack/spack/main.py", line 432, in _invoke_command
    return_val = command(parser, args)
  File "/Users/mcfadden8/spack/lib/spack/spack/cmd/view.py", line 213, in view
    exclude=args.exclude)
  File "/Users/mcfadden8/spack/lib/spack/spack/filesystem_view.py", line 214, in add_specs
    if all(map(self.add_standalone, standalones)):
  File "/Users/mcfadden8/spack/lib/spack/spack/filesystem_view.py", line 268, in add_standalone
    self.merge(spec)
  File "/Users/mcfadden8/spack/lib/spack/spack/filesystem_view.py", line 295, in merge
    raise MergeConflictError(conflicts[0])
llnl.util.link_tree.MergeConflictError: Package merge blocked by file: /Users/mcfadden8/.bin/darwin-highsierra-x86_64/share/info/dir
```

The information I have about the error in the error message is as follows:
``` console
$ ls -la /Users/mcfadden8/.bin/darwin-highsierra-x86_64/share/info/dir
lrwxr-xr-x  1 mcfadden8  54987  139 Aug 20 17:47 /Users/mcfadden8/.bin/darwin-highsierra-x86_64/share/info/dir -> /Users/mcfadden8/spack/opt/spack/darwin-highsierra-x86_64/clang-8.1.0-apple/binutils-2.31.1-347cmcghbl2pq4atso6l7peadcxsc5oo/share/info/dir
stat /Users/mcfadden8/.bin/darwin-highsierra-x86_64/share/info/dir/
16777220 8599259913 -rw-r--r-- 1 mcfadden8 (54987) 0 1804 "Aug 20 17:52:57 2018" "Aug 20 17:46:14 2018" "Aug 20 17:46:14 2018" "Aug 20 17:46:12 2018" 4194304 8 0 /Users/mcfadden8/.bin/darwin-highsierra-x86_64/share/info/dir/
file /Users/mcfadden8/.bin/darwin-highsierra-x86_64/share/info/dir/
/Users/mcfadden8/.bin/darwin-highsierra-x86_64/share/info/dir/: data
```

### Information on your system
``` console
$ uname -a
Darwin odinn.llnl.gov 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64
```

I am running with the packages.yaml and modules.yaml files from the github repository (unmodified).

**Comment by healther** [comment]
There are a number of similar issues (#6266, #7001 are the ones that I found), although I haven't seen the output of it with `-sd` so I might be wrong.

Tl;dr: `spack view` merges the content of all directories of all specs, if multiple specs install the same file view treats this as an error and errors out. 

This typically happens for either header files or drop in replacement scripts which are just copied and (hopefully) semantically identical. The example we ran into was `$prefix/lib/python2.7/site-packages/six.py` from `py-six` and `llvm`. While semantically identical, the `llvm` version had a couple of new line breaks, so we only implemented a "just ignore all conflicts and take the first file that you find" solution with the `-i` option for `spack view symlink -i`.

Can you try this and if this doesn't help, can you provide the output without `-sd`?

**Comment by mcfadden8** [comment]
Thanks @healther, using ```-i``` has helped me work around this issue.  Since I have a workaround, should I close this issue?  Or should we leave it open in order to get it to work without having to specify ```-i```?

**Comment by healther** [comment]
That depends: If you have a suggestion on how we could/should implement a heuristic, we could take this issue as a starting point for a discussion of that. 
The problem with using `-i` is that it just blindly takes the "first" occurrence. But as this worked in practice for anyone who ran into it, figuring out a permanent solution isn't a high priority for anyone. #9035 adds the `-i` suggestion in case of such an error so that a user has a higher chance of finding it.

**Comment by mcfadden8** [comment]
Closing this for now as the ```-i``` seems sufficient.  

Thanks for the assitance!
